invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessLinks/AccessDropdown.js

```diff
@@ -26,15 +28,27 @@ export class AccessDropdown extends Component {
     this.cancellableAction && this.cancellableAction.cancel();
   }
 
+  onSuccess = () =>
+    this.setState({ loading: false, actionSuccess: true, error: undefined });
+
   handleUpdate = async (permission) => {
     const { updateEndpoint } = this.props;
     const data = { permission: permission };
     this.setState({ loading: true, actionSuccess: false });
     this.cancellableAction = withCancel(http.patch(updateEndpoint, data));
 
     try {
+      const requestStartTime = Date.now();
       await this.cancellableAction.promise;
-      this.setState({ loading: false, actionSuccess: true, error: undefined });
+      const requestEndTime = Date.now();
+      const requestDuration = requestEndTime - requestStartTime;
+
+      // prolong loading for better understanding that the action was completed
```

Comment:
small suggestion, for me the comment is a bit vague because we don't know who should understand the action was completed
```suggestion
      // prolong loading to make sure UI displays to the end user for long enough that the action was completed
```